src/DataSourceProvider.vue

```diff
@@ -197,9 +200,7 @@ export default {
         });
       });
 
-      includedAccessTypes = _.uniq(includedAccessTypes);
-
-      if (includedAccessTypes.length !== this.widgetData.accessRules[0].type.length) {
+      if (this.widgetData.accessRules.length && includedAccessTypes.length !== this.widgetData.accessRules[0].type.length) {
```

Comment:
This line might fail if `accessRules` is an empty array. Is this something that could happen?

Reply:
No it can't, I tested this case the first check already returns false

Reply:
Which check? Can you point me to it?

Reply:
@squallstar 
&& finds the first false value, so if `this.WidgetData.accessRules.length` is 0 it will be false. If the result is false, stops and returns the original value of this operand.

Reply:
I see, thanks I missed that first part of the line. I've approved the PR now.